components/io_expander/esp_io_expander/esp_io_expander.c

```diff
@@ -33,23 +30,25 @@ static char *TAG = "io_expander";
 static esp_err_t write_reg(esp_io_expander_handle_t handle, reg_type_t reg, uint32_t value);
 static esp_err_t read_reg(esp_io_expander_handle_t handle, reg_type_t reg, uint32_t *value);
 
-esp_err_t esp_io_expander_set_dir(esp_io_expander_handle_t handle, uint8_t pin_num, bool is_output)
+esp_err_t esp_io_expander_set_dir(esp_io_expander_handle_t handle, uint32_t pin_num_mask, esp_io_expander_dir_t direction)
 {
-    ESP_RETURN_ON_FALSE(
-        IS_VALID_PIN_NUM(pin_num, handle), ESP_ERR_INVALID_ARG,
-        TAG, "Invalid pin_num(%d), max(%d)", pin_num, PIN_NUM_MAX(handle));
+    ESP_RETURN_ON_FALSE(handle, ESP_ERR_INVALID_ARG, TAG, "Invalid handle");
+    if (pin_num_mask >= BIT(VALID_IO_COUNT(handle))) {
+        ESP_LOGW(TAG, "Pin num mask out of range, bit higher than %d won't work", VALID_IO_COUNT(handle) - 1);
+    }
```

Comment:
If the `pin_num_mask` equals `0xFF00_0000` its value is greater then `BIT(32)`, which will print the warning on next line.

This condition is present on multiple places in this file, please check those too.

Reply:
You're right. I should use `BIT64` instead of `BIT`. These conditions have been fixed in the new [commit](https://github.com/espressif/esp-bsp/pull/86/commits/d9cc48fb6b5929df2fcab8be9aa1952b69472ea6).